### PR TITLE
DB migration for editable about page and notes field

### PR DESCRIPTION
**Pull request by zaaakk**

### CODE REVIEW

File: rome_app/views.py [hunk 1]
```diff
@@ -59,6 +59,18 @@ def index(request):
     context=std_context(request.path, style="rome/css/home.css")
     return render(request, 'rome_templates/index.html', context)
 
+def about(request):
+    context = std_context(request.path, style="rome/css/links.css")
+    try:
+        about = Static.objects.get(title="About")
+    except ObjectDoesNotExist:
+        print "nope"
```
Comment (bcail) [inline rome_app/views.py]: remove print statement

File: rome_app/views.py [hunk 2]
```diff
@@ -59,6 +59,18 @@ def index(request):
...
+        return HttpResponseNotFound('Static %s Not Found' % About)
```
Comment (bcail) [inline rome_app/views.py]: About should be 'About'

File: rome_app/views.py [hunk 3]
```diff
@@ -59,6 +59,18 @@ def index(request):
...
+    context['about_text']=about.text
+    context['about_title']=about.title
```
Comment (bcail) [inline rome_app/views.py]: should be spaces around the '=': eg. context['about_text'] = about.text

File: rome_app/static/rome/css/essays.css [hunk 4]
```diff
@@ -1,8 +1,7 @@
 
 
 #page_body ul{
-	column-count:4;
-	break-inside: avoid-column;
+
```
Comment (bcail) [inline rome_app/static/rome/css/essays.css]: can this block be removed now?

File: rome_app/views.py [hunk 5]
```diff
@@ -59,6 +59,17 @@ def index(request):
     context=std_context(request.path, style="rome/css/home.css")
     return render(request, 'rome_templates/index.html', context)
 
+def about(request):
+    context = std_context(request.path, style="rome/css/links.css")
+    try:
+        about = Static.objects.get(title="About")
+    except ObjectDoesNotExist:
+        return HttpResponseNotFound('Static %s Not Found' % 'About')
```
Comment (bcail) [inline rome_app/views.py]: don't need the % at all - can just make it 'Static About Not Found'

File: rome_app/templates/rome_templates/about.html [hunk 6]
```diff
@@ -26,12 +27,9 @@
...
-
+    <h2>{{about_title}}</h2>
+    {{about_text|markdown}}
+   
     <h2>People</h2>
     <p>Evie Lincoln (History of Art and Architecture), Faculty Lead</p>
     <p>Brian Repetto (GS, History of Art and Architecture)</p>
```
Comment (bcail) [inline rome_app/templates/rome_templates/about.html]: can we put these "People" lines in the DB markdown as well? Also, do you have markdown in your local database that you're testing with? If so, could you please email it to me and I'll put it in the test server DB when I deploy?

**Comment by bcail** [comment]
Thanks for your updates. Do you see the Travis test failure? You'll have to update the failing test to put some markdown for the about page in the database before running the test. It doesn't need to be the whole text, though. Just a bit of it. Let me know if you have questions.

File: unit_tests/test_views.py [hunk 7]
```diff
@@ -27,9 +27,10 @@ def test_index(self):
         self.assertContains(response, 'The Theater that was Rome')
 
     def test_about(self):
+        models.Static.objects.create(title='About', text='### Red Sox lineup[^n1]\n\n[^n1]: footnote text')
         response = self.client.get(reverse('about'))
         self.assertEqual(response.status_code, 200)
-        self.assertContains(response, '<h2>About</h2>')
+        self.assertContains(response, 'Red Sox')
```
Comment (bcail) [inline unit_tests/test_views.py]: Great, tests pass now. Could you add the following lines as well? They help verify that the markdown is rendered properly.
        self.assertContains(response, '<h3>Red Sox lineup') #make sure that basic markdown was rendered
        self.assertContains(response, '<p>footnote text') #make sure that footnote was rendered

File: rome_app/views.py [hunk 8]
```diff
@@ -497,6 +505,7 @@ def essay_detail(request, essay_slug):
     context=std_context(request.path, style="rome/css/essays.css")
     context['essay_text'] = essay.text
     context['essay'] = essay
+    context['author'] = essay.author
```
Comment (bcail) [inline rome_app/views.py]: You already have the essay object in the context. Could you take out this separate "author" context item, and then in the essay_detail template above, you can just use "essay.author" instead of "author".

File: unit_tests/test_views.py (hunk 7 above)
Comment (zaaakk) [reply unit_tests/test_views.py]: Could you paste that into a code block? I think github formatted the markdown.

Comment (bcail) [reply unit_tests/test_views.py]: `self.assertContains(response, 'Red Sox lineup') #make sure that basic markdown was rendered

self.assertContains(response, 'footnote text') #make sure that footnote was rendered`

File: rome_app/templates/rome_templates/essay_detail.html [hunk 9]
```diff
@@ -195,19 +214,25 @@
 <h3>Books and Prints</h3>
 <div id="relatedworks">
 {% for work in related_list %}
-    <div class="work">
+   
+    {% if work.ppid %}
```
Comment (bcail) [inline rome_app/templates/rome_templates/essay_detail.html]: Looks like you don't want to render anything if work.ppid is True: you could change this to just {% if not work.ppid %} and not use the else.

File: unit_tests/test_views.py [hunk 10]
```diff
@@ -27,9 +27,11 @@ def test_index(self):
         self.assertContains(response, 'The Theater that was Rome')
 
     def test_about(self):
+        models.Static.objects.create(title='About', text='### Red Sox lineup[^n1]\n\n[^n1]: footnote text')
         response = self.client.get(reverse('about'))
         self.assertEqual(response.status_code, 200)
-        self.assertContains(response, '<h2>About</h2>')
+        self.assertContains(response, 'Red Sox lineup') #make sure that basic markdown was rendered
```
Comment (bcail) [inline unit_tests/test_views.py]: Sorry, let me try again. The assertContains text should be
`'<h3>Red Sox lineup'`
(with the h3 element).

File: unit_tests/test_views.py [hunk 11]
```diff
@@ -27,9 +27,11 @@ def test_index(self):
...
+        self.assertContains(response, 'footnote text') #make sure that footnote was rendered
```
Comment (bcail) [inline unit_tests/test_views.py]: this should have a p element:
`'<p>footnote text'`

File: rome_app/templates/rome_templates/essay_list.html [hunk 12]
```diff
@@ -120,9 +127,35 @@
...
+    <li value class="row"><span>{{essay.author}}, <a href="{% url 'specific_essay' essay.slug %}">{{essay.title}}</a></span>
+    <br> {{essay.slug}} <br> 
+    {{essay.preview}}
+     <div class = "related">
+      {% if essay.related_list %}
+      {% for work in essay.related_list %}
+       {% if not work.ppid %}
+    
+      {% endif %}
+      {% endfor %}
```
Comment (bcail) [inline rome_app/templates/rome_templates/essay_list.html]: Lines 138-142 don't seem to do anything - they can be removed if they're not necessary.

**Comment by bcail** [comment]
@zaaakk build failed. Looks like another line needs to be removed.

File: rome_app/templates/rome_templates/essay_list.html [hunk 13]
```diff
@@ -120,9 +127,32 @@
...
-    <li class="row"><span><a href="{% url 'specific_essay' essay.slug %}">{{essay.title}}</a></span>
-    <br> {{essay.slug}} <br> </li>
+    {% if essay.is_note == False %}
+    <li value class="row"><span>{{essay.author}}, <a href="{% url 'specific_essay' essay.slug %}">{{essay.title}}</a></span>
+    <br> {{essay.slug}} <br> 
+    {{essay.preview}}
+     <div class = "related">
+      {% if essay.related_list %}
+      {% for work in essay.related_list %}
+      {% endfor %}
```
Comment (bcail) [inline rome_app/templates/rome_templates/essay_list.html]: Looks like these 2 lines don't do anything...
